client/state/themes/selectors/get-recommended-themes-filter.js

```diff
@@ -1,16 +1,13 @@
-import isSiteUsingCoreSiteEditor from 'calypso/state/selectors/is-site-using-core-site-editor';
-import 'calypso/state/themes/init';
+import { isCoreFSEEligible } from 'calypso/state/gutenberg-fse-settings/selectors';
 
 /**
  * We currently support two types of theme filtering for recommended themes,
  * with the possibility of more in the future, which will require some
  * further work in this selector.
  *
- * @param {object}      state  Global state tree
  * @param {number|null} siteId Site ID
  * @returns {string}           The string to filter on in an API request.
  */
-export function getRecommendedThemesFilter( state, siteId ) {
-	const isUsingSiteEditor = isSiteUsingCoreSiteEditor( state, siteId );
-	return isUsingSiteEditor ? 'block-templates' : 'auto-loading-homepage';
+export function getRecommendedThemesFilter( siteId ) {
+	return isCoreFSEEligible( siteId ) ? 'block-templates' : 'auto-loading-homepage';
```

Comment:
`getRecommendedThemesFilter` is no longer a valid Redux selector and can't be reliably used in a `connect` call. Why? Because the Redux store doesn't know that the underlying query has updated. In this code:
```js
connect( ( state ) => {
  const siteId = getSelectedSiteId( state );
  return {
    eligible: isCoreFSEEligible( siteId ),
  };
} );
```
the `mapStateToProps` callback will not be called again when the `isCoreFSEEligible` value changes because Redux doesn't know that. Because the Redux state didn't change. And the React component won't be updated.

Right now it probably works because there is the `isLoadingCoreFSESettings` prop that changes but if you ever removed that, the `filter` won't update.